pymc3/distributions/dist_math.py

```diff
@@ -67,6 +68,13 @@ def bound(logp, *conditions, **kwargs):
     -------
     logp with elements set to -inf where any condition is False
     """
+
+    # If called inside a model context, see if bounds check is disabled
+    model = modelcontext(kwargs.get("model"))
+    if model is not None:
```

Comment:
You can use short-circuiting behavior here safely: 
```suggestion
    if model is not None and model.disable_bounds_check:
```